scripts/portfolio.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var projects = [];
+
+function Project (dataIn) {
+  // constructor function for a Project object
+  this.projectTitle   = dataIn.projectTitle;
+  this.projectSummary = dataIn.projectSummary;
+  this.projectDetails = dataIn.projectDetails;
+  this.projectLink    = dataIn.projectLink;
+  this.projectSrcLink = dataIn.projectSrcLink;
+};
+
+Project.prototype.toHtml = function() {
+  // generate the HTML for the Project object to insert into the DOM (in an article tag).
+  var $newArticle = $('article.project-template').clone();
+  $newArticle.find('summary').text(this.projectSummary);
+  $newArticle.find('p.project-details').text(this.projectDetails);
+  var linkItem = this.projectLink;
+  if (linkItem !== null) {
```

Comment:
Nice job checking for a property before using it.  This could be a fair bit more succinct:
```
if (this.projectLink)
  $newArticle.find('a.project-link').attr('href', this.projectLink).attr('target', '_blank');
```
If your `if` clause is a single line, you can omit `{}`. Also, this saves you from needing the linkItem var.  
and using `if (something !== null)` can just be shortened to `if (something)`. which will fail if `something` is non-truthy.